python/helpers/pycharm/_jb_pytest_runner.py

```
@@ -8,11 +9,17 @@
 from _jb_runner_tools import jb_start_tests, jb_patch_separator, jb_doc_args
 from teamcity import pytest_plugin
 
+def set_up_django_environ():
+    import django
+    # TODO pass PYCHARM_DJANGO_SETTINGS_MODULE into py.test runner if enable django support
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", os.environ.get('PYCHARM_DJANGO_SETTINGS_MODULE', "settings"))
+    django.setup()
 
 
 if __name__ == '__main__':
     path, targets, additional_args = jb_start_tests()
     sys.argv += additional_args
+    set_up_django_environ()
```

I think there needs to be some detection here to ascertain if Django support is enabled in JetBrains or not.

I can think of a number of ways to do this in Python but all of them involve blindly ignoring errors, which isn't helpful.

For instance, one could do the following;

```python
try:
    import django
except ImportError:
    pass
else:
    django.setup()
```

But it would be much better if this was done using the "Enable Django Support" setting in PyCharm.